api/auth.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+message AuthAction {
+
+  enum VerifyType {
+    option allow_alias = true;
+
+    DEFAULT = 0;
+    WHITE_LIST = 0;
+    BLACK_LIST = 1;
+  }
+
+  // Check peer identity using X.509 certificate.
+  message X509 {
```

Comment:
Any reason why you aren't re-using `CertificateValidationContext` here?

Reply:
thanks for point that out.
i thought we only need to check some field of a already validated cert, and leave cert validation to listener/cluster.

i can refactor the cert field out to reuse it.